lighthouse-core/lib/stack-collector.js

```diff
@@ -15,45 +15,51 @@
 
 const fs = require('fs');
 const libDetectorSource = fs.readFileSync(
-  require.resolve('js-library-detector/library/libraries.js'),
-  'utf8'
-);
+  require.resolve('js-library-detector/library/libraries.js'), 'utf8');
+
+/**
+ * @typedef JSLibraryDetectorTest
+ * @property {string} icon Essentially an id, useful if no npm name is detected.
+ * @property {string} url
+ * @property {string|null} npm npm module name, if applicable to library.
+ * @property {function(Window): false|{version: string|null} | Promise<false|{version: string|null}>} test Returns false if library is not present, otherwise returns an object that contains the library version (set to null if the version is not detected).
```

Comment:
this is kinda a confusing type to read :) could we do something like

```js
/** @typedef {false|{version:string|null}} JSLibraryDetectorTestResult */
/** ... 
  @property {function(Window): (JSLibraryDetectorTestResult|Promise<JSLibraryDetectorTestResult>)}
```

Reply:
yah